### ISSUE TITLE
Review all DOI assign/reassign/clear functions in a targeted UI/UX session

### ISSUE THREAD

**Issue by jmacgreg** [pkp/pkp-lib#5514]
This would be for **_after_** the OJS 3.2. release. The Reassign/Assign DOI functions just aren't clear enough, even for me and @AhemNason, to understand exactly what the implications are for pressing them. Likewise, the Clear DOI button function for any individual DOI. We click them, and then spend 5 minutes trying to figure out what happened and what we should expect. Just marking this for future targeted review.

**Comment by NateWr** [comment]
@jmacgreg and @AhemNason I know you've both got plenty on your plate. But I've assigned this one to you two. Can we hand this off to you and have you come back to us with some recommended changes?

**Comment by AhemNason** [comment]
I'd be happy to be involved, that's for sure. Maybe we should pull in @israelcefrin as well, though?

**Comment by eczheng** [comment]
Hi everyone, I've been testing this from the user's perspective for a few weeks (using 3.2.1.1). 

From what I've seen, the most significant change is that new Submissions are not being **auto**-assigned a DOI. Before 3.2, DOIs were automatically assigned prior to the production stage. After 3.2, this automatic assignment is not happening (or at least, not consistently...) Editors now have to manually click on _Publication > Identifiers > Assign_. If they don't do that first, a warning message will display when they attempt to publish, but that can be easily overridden. For editors who are unaccustomed to this step, it's pretty easy to overlook.

Within the plugin, the Assign and Clear functions seem to be working as expected (I'm not familiar with how Reassign worked prior to 3.2, so I haven't test it much). FYI: the Clear function only works if you unpublish the article first -- otherwise, the Save button is locked.

**Comment by AhemNason** [comment]
Thanks for this @eczheng. I've also noticed this is a bit of an issue.

**Comment by NateWr** [comment]
So for your use-case, ideally you want to block publication when a DOI has not been assigned. Is that right? This would be fairly easy to accomplish with a plugin. Setting such hard boundaries in the application is hard, because each journal has their own DOI regime. 

That said, I think that we're considering moving DOI assignment out of the publishing process altogether and either automating it or moving it to an assign/deposit UI that happens independently of the schedule for publication workflow.

**Comment by AhemNason** [comment]
It's more that when the DOI plugin is set to "assign DOIs automatically", you still have to click the "assign" button on the identifier part of the workflow in order for it to assign. This isn't really automatic. If it's set to automatic, I should be able to just skip that assignment step without worrying about _no DOI_ getting assigned.

**Comment by eczheng** [comment]
Our use cases would concur with @AhemNason. I don't think blocking publication without DOI is a good solution. I can imagine that some items wouldn't need a DOI for any number of reasons.

I would add: Since many journals include the DOI in their galleys, they would expect to know the DOI several stages prior to Publication (during copyediting or layout). Since the DOI is generated based on the Submission ID, it might make sense to populate the Identifier field at the time of Submission (similar to how it was done prior to 3.2). As long as they never become public, I think it's fine to have unregistered DOIs.

**Comment by NateWr** [comment]
Hmm, I'm not seeing an option in the DOI plugin settings to "assign DOIs automatically". Perhaps this has been removed?

This is another area where we've backed ourselves into a corner by giving so many options for the DOI pattern. We don't necessarily have all the information we need to assign a DOI during submission and in some cases won't even have that information when publishing.

I'm not sure but I think that we're planning to drop those DOI pattern options in the future, which should make it possible for us to do a lot more of his automatically.

**Comment by eczheng** [comment]
You're right, @NateWr, I didn't realize that the settings don't say "automatic" anywhere. I think it was just expected behaviour, based on pre-3.2.

Hm... yes, the DOI pattern options that construction DOIs from Volume / Issue numbers are complicating... One of these complications is a confusing loop that some editors may get stuck in:
1. Trying to schedule an issue, the warning: _A DOI has not been assigned to this publication._
2. Trying to assign a DOI, the warning:  _You can not generate a DOI until this publication has been assigned to an issue._

The (unintuitive) solution would be to override warning #1 and assign to Issue, then go to Identifier > Assign DOI and save. 

For rolling publications (journals which add articles to already-published issues for immediate publication): editor has to assign the issue, then exit out of the wizard when the DOI warning pops up. The issue metadata will still save, so they can assign DOI. Alternatively, they can unpublish, assign DOI, then republish. For now, are those the only options?

**Comment by israelcefrin** [comment]
@ewhanson isn't this issue covered in your current work with DOI management?

**Comment by ewhanson** [comment]
@israelcefrin, yes, this is covered in the scope of the current work on DOI management. 

The plan is roughly to create an "automatic first" approach to DOI creation. 

1. DOIs will be automatically created and assigned.
2. If users need to edit the DOI, they will be able to do so from a central DOI management page. Here the option will be to edit a DOI rather than assign/reassign. 
3. Likewise the language around clearing DOIs will be removed.

@eczheng, the challenge regarding DOI patterns constructed from Volume/Issue numbers will also be addressed as part of the DOI management overhaul, but this hasn't been implemented yet. I can add a link to the relevant issue/more details once that is further along.
